src/CompilerGlobals.cpp

```diff
@@ -0,0 +1,43 @@
+#include "CompilerGlobals.h"
+
+#include "Util.h"
+
+namespace Halide {
+namespace Internal {
+
+namespace {
+
+Globals the_real_globals;
+
+size_t compute_stack_size() {
```

Comment:
I think the compiler stack size doesn't belong in this struct. It's not mutable state - it's configuration.

Reply:
It can be changed by `set_stack_size()` at runtime.

Reply:
But it's not mutated as part of a compilation of a Halide pipeline. It's libHalide configuration state rather than Generator state. Someone might feasibly want to set it with code outside of a loop that calls compile_multitarget a bunch of times.

Reply:
Fair. Will change.